.github/workflows/infrastructure.yml

```diff
@@ -0,0 +1,173 @@
+# This GHA is responsible for `APIM` supporting infrastructure
+# creation and configuration using `az cli` bash scripting.
+#
+# Standard Azure naming convention has been followed:
+# https://learn.microsoft.com/en-us/azure/cloud-adoption-framework/ready/azure-best-practices/resource-naming
+#
+#
+# Following Azure services are consumed:
+# 1. Azure resource group - https://learn.microsoft.com/en-us/cli/azure/group?view=azure-cli-latest#az-group-create
+# 2. Azure container registry - https://learn.microsoft.com/en-us/cli/azure/acr?view=azure-cli-latest#az-acr-create
+# 3. Azure container app environment - https://learn.microsoft.com/en-us/azure/container-apps/environment
+# 4. Azure container app - https://learn.microsoft.com/en-us/azure/container-apps/containers
+# 
+#
+# Execution
+# *********
+# GHA is only invoked when following conditions are satisfied:
+# 1. Push to the `infrastructure` branch only.
+# 2. Exact file path matches `.github/workflows/infrastructure.yml`.
+
+name: Infrastructure 🔨
+run-name: APIM base infrastructure build from ${{ github.repository }}
+
+on:
+  push:
+    branches: [ APIM-149-ia-c-dev-environment ]
+    paths: [ .github/workflows/infrastructure.yml ]
+
+env:
+  product: apim
+  environment: infrastructure
```

Comment:
Isn't `environment` refering to Dev/test/prod?

Reply:
Nope since this is `infrastructure` which environment to deploy will be dictated by an environment variable configured in GitHub.